### PR TITLE
Add section on keeping CI green

### PR DESCRIPTION
**Pull request by ambv**
----
:books: Documentation preview :books:: https://cpython-devguide--1215.org.readthedocs.build/

### CODE REVIEW

**Comment by ambv** [comment]
Screenshot 2023-11-03 at 16 56 13

File: getting-started/pull-request-lifecycle.rst [hunk 1]
```diff
@@ -487,14 +487,55 @@ Instead of simply "approving" the pull request, leave comments.  For example:
...
+---------------------------------------------
 
 A core developer can dismiss another core developer's review if they confirmed
 that the requested changes have been made.  When a core developer has assigned
 the PR to themselves, then it is a sign that they are actively looking after
 the PR, and their review should not be dismissed.
 
+.. _keeping-ci-green:
+
+Keeping Continuous Integration green
```
Comment (hugovk) [inline getting-started/pull-request-lifecycle.rst:503]:
```suggestion
Keeping continuous integration green
```

File: getting-started/pull-request-lifecycle.rst [hunk 2]
```diff
@@ -487,14 +487,55 @@ Instead of simply "approving" the pull request, leave comments.  For example:
...
+====================================
+
+Our change management workflows generally won't allow merging PRs with
+failures on them. Therefore, if you see a CI failure on a PR, have a look
```
Comment (hugovk) [inline getting-started/pull-request-lifecycle.rst:507]:
```suggestion
failures. Therefore, if you see a CI failure on a PR, have a look
```

File: getting-started/pull-request-lifecycle.rst [hunk 3]
```diff
@@ -487,14 +487,55 @@ Instead of simply "approving" the pull request, leave comments.  For example:
...
+Usually the failure will be directly related to the changes in the current
+PR. If you happen to have any insight into the failure, let the author know
+in a review comment. CI runs sometimes generate thousands of lines of output.
+Even something as simple as finding the traceback and putting it in the
+comment will be helpful to the PR author.
+
+If the failure doesn't look related to the change you're looking at, check
+if it's not present on the `Release Status`_ Buildbot dashboard as well.
+If so, that means the failure was introduced in a prior change. Using Buildbot's
+UI you can find which PR introduced the issue and comment there that it
```
Comment (hugovk) [inline getting-started/pull-request-lifecycle.rst:519]:
```suggestion
UI you can find which PR introduced the issue and comment that it
```

File: getting-started/pull-request-lifecycle.rst [hunk 4]
```diff
@@ -487,14 +487,55 @@ Instead of simply "approving" the pull request, leave comments.  For example:
...
+If you still don't see where the failure originates from, check for
+an "This branch is out-of-date with the base branch" sign next to the
+list of executed checks. Clicking "Update branch" next to this message
+will merge in the latest changes from the base branch into the PR.
+
+If this still doesn't help with the failure on the PR, you can try
+to re-run that particular failed check. Go to the red GitHub Action job,
+and find a button on the top right called "Re-run jobs". Select
+"Re-run failed jobs". The button will only be present when all other jobs
+finished running.
```
Comment (hugovk) [inline getting-started/pull-request-lifecycle.rst:531]: Only applies to core devs and triagers?

File: getting-started/pull-request-lifecycle.rst [hunk 5]
```diff
@@ -487,14 +487,55 @@ Instead of simply "approving" the pull request, leave comments.  For example:
...
+comment will be helpful to the PR author.
+
+If the failure doesn't look related to the change you're looking at, check
+if it's not present on the `Release Status`_ Buildbot dashboard as well.
+If so, that means the failure was introduced in a prior change. Using Buildbot's
+UI you can find which PR introduced the issue and comment there that it
+affects other PRs.
+
+If you still don't see where the failure originates from, check for
+an "This branch is out-of-date with the base branch" sign next to the
```
Comment (JelleZijlstra) [inline getting-started/pull-request-lifecycle.rst:523]:
```suggestion
a "This branch is out-of-date with the base branch" sign next to the
```

File: getting-started/pull-request-lifecycle.rst (hunk 4 above)
Comment (JelleZijlstra) [reply getting-started/pull-request-lifecycle.rst:531]: That's also true for the previous paragraph (about merging in main).

File: getting-started/pull-request-lifecycle.rst (hunk 1 above)
Comment (ambv) [reply getting-started/pull-request-lifecycle.rst:503]: but that's CI, a proper name, like GitHub?

File: getting-started/pull-request-lifecycle.rst (hunk 4 above)
Comment (hugovk) [reply getting-started/pull-request-lifecycle.rst:531]: Any contributor can merge `main` into their PR, for example: https://github.com/python/cpython/pull/105984#commits-pushed-3cd8d82

File: getting-started/pull-request-lifecycle.rst (hunk 1 above)
Comment (hugovk) [reply getting-started/pull-request-lifecycle.rst:503]: No, it's a common noun.

If short for continuous integration server/service, there are many named ones: Azure Pipelines, Jenkins, Gitlab Pipelines, AppVeyor. The proper noun of a CI service we're using is GitHub Actions.

If continuous integration as the name for a practice, it's also a common noun. For example, see https://en.wikipedia.org/wiki/Continuous_integration

File: getting-started/pull-request-lifecycle.rst [hunk 6]
```diff
@@ -487,14 +487,55 @@ Instead of simply "approving" the pull request, leave comments.  For example:
...
+list of executed checks. Clicking "Update branch" next to this message
+will merge in the latest changes from the base branch into the PR.
+
+If this still doesn't help with the failure on the PR, you can try
+to re-run that particular failed check. Go to the red GitHub Action job,
+and find a button on the top right called "Re-run jobs". Select
```
Comment (ezio-melotti) [inline getting-started/pull-request-lifecycle.rst:529]:
```suggestion
click on the "Re-run jobs" button on the top right, and select
```

**Pull request merged**